### PR TITLE
chore(wrangler): Add `pages dev` tests

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**
## What this PR solves / how to test

This PR adds some sanity-check unit and e2e tests for `wrangler pages dev`.

Screenshot 2024-03-22 at 13 33 27

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: not user facing. Only adding tests
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: not user facing. Only adding tests

### CODE REVIEW

File: packages/wrangler/e2e/pages-dev.test.ts [hunk 1]
```diff
@@ -5,36 +5,87 @@ import getPort from "get-port";
 import shellac from "shellac";
 import { fetch } from "undici";
 import { beforeEach, describe, expect, it } from "vitest";
+import { normalizeOutput } from "./helpers/normalize";
 import { retry } from "./helpers/retry";
 import { dedent, makeRoot, seed } from "./helpers/setup";
 import { WRANGLER } from "./helpers/wrangler-command";
 
+type MaybePromise<T = void> = T | Promise<T>;
```
Comment (CarmenPopoviciu) [inline packages/wrangler/e2e/pages-dev.test.ts:13]: technically, all of `MaybePromise`, `SessionData`, `waitUntilOutputContains` and `runDevSession` are duplicated from wrangler proper. The only diff is the `dev` command `runDevSession` runs under the hood (`wrangler dev` vs `wrangler pages dev`).

We could keep things as they are, in the idea that convergence will make one of these obsolete, or we could extract the common functionality in a helper to clean up the code duplication. No major blocker, just calling it out

**Comment by CarmenPopoviciu** [comment]
Ironically, the `E2E Tests` CI job is skipped, so y'all just have to take my word that all these tests are 🍏  ;)

File: packages/wrangler/e2e/pages-dev.test.ts [hunk 2]
```diff
@@ -227,4 +574,137 @@ describe("pages dev tests", () => {
 			"Unexpected error building Functions directory"
 		);
 	});
+
+	it("should modify _routes.json during dev session (_worker)", async () => {
```
Comment (petebacondarwin) [inline packages/wrangler/e2e/pages-dev.test.ts:578]: Is this message correct? 
Shouldn't it be "should support modifying ..."?

File: packages/wrangler/src/__tests__/pages/dev.test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { runInTempDir } from "../helpers/run-in-tmp";
+import { runWrangler } from "../helpers/run-wrangler";
+
+/**
+ * Given that `runWrangler()` mocks out the underlying implementation
+ * (see "jest.setup.ts") there's only so much worth testing here.
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/pages/dev.test.ts:6]: I believe this can be turned off if needed. But perhaps for the future...

**Review by petebacondarwin — APPROVED** [review]

File: packages/wrangler/e2e/pages-dev.test.ts (hunk 2 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/e2e/pages-dev.test.ts:578]: yes indeed. fixed! ✅

**Pull request merged**